examples/textual_inversion/textual_inversion.py

```diff
@@ -549,18 +556,23 @@ def main():
                 # Get the text embedding for conditioning
                 encoder_hidden_states = text_encoder(batch["input_ids"])[0]
 
-                # Predict the noise residual
-                model_pred = unet(noisy_latents, timesteps, encoder_hidden_states).sample
-
                 # Get the target for loss depending on the prediction type
                 if noise_scheduler.config.prediction_type == "epsilon":
                     target = noise
                 elif noise_scheduler.config.prediction_type == "v_prediction":
                     target = noise_scheduler.get_velocity(latents, noise, timesteps)
                 else:
                     raise ValueError(f"Unknown prediction type {noise_scheduler.config.prediction_type}")
+                    
+                # Predict the noise residual
+                if args.amp_data_type == "bf16":
+                    with torch.autocast(device_type=str(accelerator.device), dtype=torch.bfloat16):
+                        model_pred = unet(noisy_latents, timesteps, encoder_hidden_states).sample
+                        loss = F.mse_loss(model_pred, target, reduction="none").mean([1, 2, 3]).mean()
+                else:
+                    model_pred = unet(noisy_latents, timesteps, encoder_hidden_states).sample
+                    loss = F.mse_loss(model_pred, target, reduction="none").mean([1, 2, 3]).mean()
```

Comment:
Instead of using `autocrats` here, we could directly cast `unet` and `van` to low precision if `mixed_precision` is turned on as it's done in dreambooth script cf https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth.py#LL566

This way we don't need to add any new argument.

Reply:
Thanks for your comments. We know that `.to` and `.autocast` are different ways since `to` convert all parameters to bf16 and `.autocast` may keeps come operations with fp32 (https://pytorch.org/docs/stable/amp.html). Therefore, the second way is more likely to keep the origin model's (fp32) accuracy.

Reply:
You are right @jiqing-feng ! But here the `unet` and `vae` are not trained and are always in inference mode, so using `.to` here shouldn't hurt the accuracy. IMO `autocast` is necessary for the model being trained, which in this case is `text_encoder` which already get's wrapped in that context in `accelerate`.